client/core/trade.go

```diff
@@ -352,7 +352,7 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
 				t.metaData.Status = order.OrderStatusExecuted
 			}
 		case order.CancelOrderType:
-			t.metaData.Status = order.OrderStatusCanceled
+			t.metaData.Status = order.OrderStatusCanceled // ?
```

Comment:
@buck54321 Some things here are confusing me that I'm hoping you can help me with.

I didn't think a `trackedTrade` itself could be a cancel order, since we have a `trackedCancel` type that goes in the  `trackedTrade.cancel` field.

Also, it seems like the assignment `t.metaData.Status = order.OrderStatusCanceled` down in the `if includesCancellation` branch comes after  t.db.UpdateOrder(t.metaOrder())` so that it won't be reflected in the DB.

Reply:
Yeah. This `case` seems unnecessary. 

> Also, it seems like the assignment `t.metaData.Status = order.OrderStatusCanceled` down in the if `includesCancellation` branch comes after `t.db.UpdateOrder(t.metaOrder())` so that it won't be reflected in the DB.

It does look that way. Seems like a bug that snuck in with #378.